File: Ejercicio 4/Ejercicio 4 (Recorrido por Columnas).java
public class Main {
    public static void main(String[] args) {
        // Declarar e inicializar una matriz 3x3
        int[][] matriz = {
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        };
        
        System.out.println("Matriz 3x3 original:");
        // Mostrar la matriz original
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                System.out.print(matriz[i][j] + " ");
            }
            System.out.println();
        }
        
        System.out.println("\nRecorrido por columnas:");
        // Recorrido por columnas
        for (int columna = 0; columna < 3; columna++) {
            System.out.print("Columna " + (columna + 1) + ": ");
            for (int fila = 0; fila < 3; fila++) {
                System.out.print(matriz[fila][columna] + " ");
            }
            System.out.println();
        }
        
        System.out.println("\nRecorrido completo por columnas:");
        // Recorrido completo columna por columna
        for (int columna = 0; columna < 3; columna++) {
            for (int fila = 0; fila < 3; fila++) {
                System.out.println("Elemento [" + fila + "][" + columna + "] = " 
                     + matriz[fila][columna]);
            }
        }
    }
}

File: Ejercicio 4/Ejercicio 4 (Recorrido por Columnas).js
// Declarar e inicializar una matriz 3x3
const matriz = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
];

console.log("Matriz 3x3 original:");
// Mostrar la matriz original
for (let i = 0; i < 3; i++) {
    let fila = "";
    for (let j = 0; j < 3; j++) {
        fila += matriz[i][j] + " ";
    }
    console.log(fila);
}

console.log("\nRecorrido por columnas:");
// Recorrido por columnas
for (let columna = 0; columna < 3; columna++) {
    let contenidoColumna = "Columna " + (columna + 1) + ": ";
    for (let fila = 0; fila < 3; fila++) {
        contenidoColumna += matriz[fila][columna] + " ";
    }
    console.log(contenidoColumna);
}

console.log("\nRecorrido completo por columnas:");
// Recorrido completo columna por columna
for (let columna = 0; columna < 3; columna++) {
    for (let fila = 0; fila < 3; fila++) {
        console.log("Elemento [" + fila + "][" + columna + "] = " + matriz[fila][columna]);
    }
}